ma_gym/envs/lumberjacks/lumberjacks.py

```diff
@@ -0,0 +1,351 @@
+        """
+        if agent_id is not None:
+            assert agent_id <= self.n_agents
+            return [k.upper() for k, v in sorted(ACTIONS_IDS.items(), key=lambda item: item[1])]
+        else:
+            return [[k.upper() for k, v in sorted(ACTIONS_IDS.items(), key=lambda item: item[1])]]
+
+    def reset(self) -> List[List[float]]:
+        self._init_episode()
+        self._step_count = 0
+        self._total_episode_reward = np.zeros(self.n_agents)
+        self._agent_dones = [False] * self.n_agents
+
+        return self.get_agent_obs()
+
+    def _init_episode(self):
+        """Initialize environment for next episode.
+
+        Fills `self._agents`, self._agent_map` and `self._tree_map` with new values.
+        """
+        init_possitions = self._generate_init_pos()
+        agent_id, tree_id = 0, self.n_agents
+        self._agents = []
+        self._agent_map = np.zeros((
+            self._grid_shape[0] + 2 * (self._agent_view[0]),
+            self._grid_shape[1] + 2 * (self._agent_view[1]),
+            self.n_agents
+        ), dtype=np.int32)
+        self._tree_map = np.zeros((
+            self._grid_shape[0] + 2 * (self._agent_view[0]),
+            self._grid_shape[1] + 2 * (self._agent_view[1]),
+        ), dtype=np.int32)
+
+        for pos, cell in np.ndenumerate(init_possitions):
+            pos = self._to_extended_coordinates(pos)
+            if cell == PRE_IDS['agent']:
+                self._agent_map[pos[0], pos[1], agent_id] = 1
+                self._agents.append(Agent(agent_id, pos=pos))
+                agent_id += 1
+            elif cell == PRE_IDS['tree']:
+                self._tree_map[pos] = np.random.randint(1, self.n_agents + 1)
+                tree_id += 1
+
+    def _to_extended_coordinates(self, relative_coordinates):
+        """Translate relative coordinates in to the extended coordinates."""
+        return (relative_coordinates[0] + self._agent_view[0], relative_coordinates[1] + self._agent_view[1])
+
+    def _to_relative_coordinates(self, extended_coordinates):
+        """Translate extended coordinates in to the relative coordinates."""
+        return (extended_coordinates[0] - self._agent_view[0], extended_coordinates[1] - self._agent_view[1])
+
+    def _generate_init_pos(self) -> np.ndarray:
+        """Returns randomly selected initial positions for agents and trees in relative coordinates.
+
+        No agent or trees share the same cell in initial positions.
+        """
+        init_pos = np.array(
+            [PRE_IDS['agent']] * self.n_agents +
+            [PRE_IDS['tree']] * self._n_trees +
+            [PRE_IDS['empty']] * (np.prod(self._grid_shape) - self.n_agents - self._n_trees)
+        )
+        np.random.shuffle(init_pos)
+        return np.reshape(init_pos, self._grid_shape)
+
+    def render(self, mode='human'):
+        img = copy.copy(self._base_img)
+        cell_size = (CELL_SIZE, CELL_SIZE / 2)
```

Comment:
Why half of the cell size?

Reply:
Also, It will be nice if everything is centered.

Optional: Instead of green box, it will be nice to have a tree with reducing color gradient as it changes strength. :evergreen_tree:

Reply:
The half cell size is because I split the horizontal cell into two parts (tree part and agent part) during the rendering. But technically it's the same cell in environment logic, it just solves the issue with multiple agents/trees on the very same cell.

I can use some kind of tree image, however, I wouldn't go for the gradient colour. The number indicator seems to be more explicit and one doesn't need to compare the colour gradients.

Reply:
If agent and tree are not in the same cell, then you use center position otherwise split the cell size.

Reply:
In my opinion, this would case weird flickering, when the agents move and in the end, it would be more confusing for the user. 
About the _tree_ shape, is there any support in `utils.draw` module? Right now I would rather spend time on _Level-based foraging_. But if you insist on it, we can open an issue for this request.

Reply:
I would say give it a try as I believe this is just a couple of lines of change.
If it doesn't look nice, you can restore back to the current setting. 

For the tree, you can just keep the green color as of now as tree image is not critical. 
Also, As of now, utils.draw doesn't support the insertion of image components.

But, the trick could be to use emoji's which could be inserted just as writing text.

Reference: 
https://github.com/carpedm20/emoji
https://stackoverflow.com/questions/61024482/writing-an-emoji-on-an-image-using-opencv
https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md

Reply:
It's better than I have expected. What do you think?
![Lumberjacks-v0](https://user-images.githubusercontent.com/13243167/95125690-9fd3bb00-0755-11eb-937a-787d7b6cf724.gif)

About the emoji, it seems to be overkill for our purpose. Firstly, we would add a new dependency and secondly it would increase the complexity of resizing. But let's open issue with it and keep there reference.

Reply:
Looks good!